core/server/main.cpp

```diff
@@ -147,6 +151,42 @@ int main(int a_argc, char **a_argv) {
 
         return 0;
       }
+      if (cfg_log_level != UINT_MAX) {
+        if (cfg_log_level >=
+            static_cast<unsigned int>(LogLevel::LAST_SENTINEL)) {
+          DL_ERROR(log_context,
+                   "Invalid log level provided, defaulting to INFO");
+        } else {
+          LogLevel cast_log_level = static_cast<LogLevel>(cfg_log_level);
+          std::string str_log_level = "";
+          switch (cast_log_level) {
+          case LogLevel::CRITICAL:
+            str_log_level = "CRITICAL";
+            break;
+          case LogLevel::ERROR:
+            str_log_level = "ERROR";
+            break;
+          case LogLevel::WARNING:
+            str_log_level = "WARNING";
+            break;
+          case LogLevel::INFO:
+            str_log_level = "INFO";
+            break;
+          case LogLevel::DEBUG:
+            str_log_level = "DEBUG";
+            break;
+          case LogLevel::TRACE:
+            str_log_level = "TRACE";
+            break;
+          case LogLevel::LAST_SENTINEL:
```

Comment:
Should I throw an exception in this case?

Reply:
Yes

Reply:
Added 🫡